CMakeLists.txt

```
@@ -204,6 +204,12 @@ if(SFML_ENABLE_SANITIZERS)
     string(APPEND CMAKE_CXX_FLAGS " -fno-omit-frame-pointer -fno-sanitize-recover=all -fsanitize=undefined")
 endif()
 
+option(SFML_ENABLE_STDLIB_ASSERTIONS "Enable standard library assertions" OFF)
+if(SFML_ENABLE_STDLIB_ASSERTIONS)
+    add_definitions(-D_GLIBCXX_ASSERTIONS=1)                                   # see https://gcc.gnu.org/wiki/LibstdcxxDebugMode
+    add_definitions(-D_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_EXTENSIVE) # see https://libcxx.llvm.org/Hardening.html
```

```suggestion
    add_compile_definitions(_GLIBCXX_ASSERTIONS=1)                                   # see https://gcc.gnu.org/wiki/LibstdcxxDebugMode
    add_compile_definitions(_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_EXTENSIVE) # see https://libcxx.llvm.org/Hardening.html
```
CMake recommends using `add_compile_definitions` in lieu of `add_definitions`.

Source: https://cmake.org/cmake/help/latest/command/add_definitions.html